Framework/Algorithms/src/ReflectometryReductionOne.cpp

```diff
@@ -612,7 +630,38 @@ void ReflectometryReductionOne::exec() {
   }
 
   IvsQ = this->toIvsQ(IvsLam, correctDetectorPositions, theta, isPointDetector);
-
+  double momentumTransferMinimum = getProperty("MomentumTransferMinimum");
+  double momentumTransferStep = getProperty("MomentumTransferStep");
+  double momentumTransferMaximum = getProperty("MomentumTransferMaximum");
+  if (!isDefault("MomentumTransferMinimum") &&
+      !isDefault("MomentumTransferStep") &&
+      !isDefault("MomentumTransferMaximum")) {
+    MantidVec QParams;
+    QParams.push_back(momentumTransferMinimum);
+    QParams.push_back(-momentumTransferStep);
+    QParams.push_back(momentumTransferMaximum);
+    IAlgorithm_sptr algRebin = this->createChildAlgorithm("Rebin");
+    algRebin->initialize();
+    algRebin->setProperty("InputWorkspace", IvsQ);
+    algRebin->setProperty("OutputWorkspace", IvsQ);
+    algRebin->setProperty("Params", QParams);
+    algRebin->execute();
+    if (!algRebin->isExecuted())
+      throw std::runtime_error("Failed to run Rebin algorithm");
+    IvsQ = algRebin->getProperty("OutputWorkspace");
+  }
```

Comment:
At this point, if `MomentumTransferMinimum`, `MomentumTransferStep` and `MomentumTransferMaximum` don't have the default values, it means that either the user supplied their own values or `ReflectometryReductionOneAuto` was able to set their values using the supplied `ThetaIn` and the information (`LambdaMin`, `LambdaMax`) in the IDF. This is fine.

However, if they were not set we will have to calculate them here using the `theta` value calculated in `this->toIvsQ()`.